kolibri/plugins/device/assets/src/views/FacilitiesPage/facilityTasksQueue.js

```diff
@@ -43,8 +43,13 @@ export default {
   computed: {
     facilityIsSyncing() {
       return function isSyncing(facility) {
-        const syncTasks = this.facilityTasks.filter(t => isSyncTask(t) && !t.clearable);
-        return Boolean(syncTasks.find(task => taskFacilityMatch(task, facility)));
+        const activeSyncTasks = this.facilityTasks.filter(
+          t =>
+            isSyncTask(t) &&
+            ((t.repeat !== null && !t.clearable) ||
```

Comment:
None blocking comment: I am wondering if we could introduce intermediate variables with descriptive names to improve code readability. Something like 
```
 const isSync = isSyncTask(task);
  const isRepeatableNonClearable = task.repeat !== null && !task.clearable;
```

So that instead of using a complex logical expression, the return statement in the `filter()` method utilizes the intermediate variable

Reply:
Yes, I think this functional style can be a bit hard to read and intermediary variables can help. I mostly didn't do that because it was like that when I got here! I'll leave for now, but I think worth thinking about when we do further reworking of the frontend task handling (which I have some overall ideas on how we can migrate to a dedicated composable).